Realm/RLMQueryUtil.mm

```diff
@@ -1850,6 +1848,45 @@ bool is_self_value_for_key_path_function_expression(NSExpression *expression)
                        @"Only support compound, comparison, and constant predicates");
     }
 }
+
+// This function returns the nested subscripts from a NSPredicate with the following format `anyCol[0]['key']['all']`
+// and its respective keypath (including any linked keypath)
+// This will iterate each argument of the NSExpression and its nested NSExpressions, takes the constant subscript
+// and creates a PathElement to be used in the query.
+NSString* QueryBuilder::get_path_elements(std::vector<PathElement> &paths, NSExpression *expression) {
+    NSString *keyPath = @"";
+    for (NSUInteger i = 0; i < expression.arguments.count; i++) {
+        NSString *nestedKeyPath = @"";
+        if (expression.arguments[i].expressionType == NSFunctionExpressionType) {
+            nestedKeyPath = get_path_elements(paths, expression.arguments[i]);
+        } else if (expression.arguments[i].expressionType == NSConstantValueExpressionType) {
+            id value = [expression.arguments[i] constantValue];
+            if ([value isKindOfClass:[NSNumber class]]) {
+                paths.push_back(PathElement{[(NSNumber *)value intValue]});
+            } else if ([value isKindOfClass:[NSString class]]) {
+                NSString *key = (NSString *)value;
+                if ([key isEqual:@"*"]) {
```

Comment:
`'*'` is better, but it is still a legal key. I poked a bit at what NSPredicate will let us put here and the reasonable-seeming options are `anyCol['a'].@any['b']` and `anyCol['a'][#any]['b']` (note the lack of quotes). The latter seems better and also the parser is weird and doesn't allow `@` followed by any of the NSPredicate keywords, which happens to include "any" and "all".

This shows up in the parsed predicate as a NSKeyPathExpressionType with `[(id)expression.arguments[i] predicateFormat]` being `@"#any"` (but `keyPath` is `@"any").